src/schema/v2/artwork/deleteArtworkMutation.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  GraphQLString,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import Artwork from "schema/v2/artwork"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+
+interface DeleteArtworkMutationInputProps {
+  id: string
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "DeleteArtworkSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    artwork: {
+      type: Artwork.type,
+      resolve: (artwork) => artwork,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "DeleteArtworkFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "DeleteArtworkResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const deleteArtworkMutation = mutationWithClientMutationId<
+  DeleteArtworkMutationInputProps,
+  any,
+  ResolverContext
+>({
+  name: "DeleteArtworkMutation",
+  description: "Deletes an artwork.",
+  inputFields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The ID of the artwork to delete.",
+    },
+  },
+  outputFields: {
+    artworkOrError: {
+      type: ResponseOrErrorType,
+      description:
+        "On success: the deleted artwork. On error: the error that occurred.",
```

Comment:
This is possible because the artwork is soft-deleted, yah?

Reply:
You mean returning the deleted artwork? Yea - even if it was hard-deleted our MO is still to return the full object in the JSON from the DELETE.